middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -933,6 +939,13 @@ let binary_primitive env dbg f x y =
   | Bigarray_get_alignment align -> C.bigstring_get_alignment x y align dbg
   | Atomic_exchange -> C.atomic_exchange ~dbg x y
   | Atomic_fetch_and_add -> C.atomic_fetch_and_add ~dbg x y
+  | Poke kind ->
+    let memory_chunk =
+      K.Standard_int_or_float.to_kind_with_subkind kind
+      |> C.memory_chunk_of_kind
+    in
+    C.store ~dbg memory_chunk Assignment ~addr:x ~new_value:y
+    |> C.return_unit dbg
```

Comment:
Will the `return_unit` always be optimized out? If not it seems worthwhile to make poke return nothing (or unboxed unit) (or something like an uninitialized unboxed nativeint)